src/schemas.ts

```diff
@@ -0,0 +1,370 @@
+        'su',
+        'sv',
+        'sw',
+        'ta',
+        'te',
+        'tg',
+        'th',
+        'ti',
+        'tk',
+        'tl',
+        'tn',
+        'to',
+        'tr',
+        'ts',
+        'tt',
+        'tw',
+        'ty',
+        'ug',
+        'uk',
+        'ur',
+        'uz',
+        've',
+        'vi',
+        'vo',
+        'wa',
+        'wo',
+        'xh',
+        'yi',
+        'yo',
+        'za',
+        'zh',
+        'zu',
+        'und',
+      ],
+    },
+    preloadedDependencies: {
+      type: 'array',
+      items: {
+        type: 'object',
+        properties: {
+          machineName: { type: 'string' },
+          majorVersion: { type: ['integer', 'string'] },
+          minorVersion: { type: ['integer', 'string'] },
+        },
+        required: ['machineName', 'majorVersion', 'minorVersion'],
+      },
+    },
+    embedTypes: {
+      type: 'array',
+      minItems: 1,
+      maxItems: 2,
+      items: { type: 'string', enum: ['div', 'iframe'] },
+      uniqueItems: true,
+    },
+    authors: {
+      type: 'array',
+      items: {
+        type: 'object',
+        properties: {
+          name: { type: 'string' },
+          role: { type: 'string', enum: ['Author', 'Editor', 'Licensee', 'Originator'] },
+        },
+        required: ['name', 'role'],
+      },
+      nullable: true, // JSON schema treats undefined as nullable?
+    },
+    source: {
+      type: 'string',
+      nullable: true, // JSON schema treats undefined as nullable?
+    },
+    license: {
+      type: 'string',
+      enum: [
+        'CC-BY',
+        'CC BY-SA',
+        'CC BY-ND',
+        'CC BY-NC',
+        'CC BY-NC-SA',
+        'CC CC-BY-NC-CD',
+        'CC0 1.0',
+        'GNU GPL',
+        'PD',
+        'ODC PDDL',
+        'CC PDM',
+        'C',
+        'U',
+      ],
+      nullable: true, // JSON schema treats undefined as nullable?
+    },
+    licenseVersion: {
+      type: 'string',
+      nullable: true, // JSON schema treats undefined as nullable?
+    },
+    licenseExtras: {
+      type: 'string',
+      nullable: true, // JSON schema treats undefined as nullable?
+    },
+    yearFrom: {
+      type: 'string',
+      nullable: true, // JSON schema treats undefined as nullable?
+    },
+    yearTo: {
+      type: 'string',
+      nullable: true, // JSON schema treats undefined as nullable?
+    },
+    changes: {
+      type: 'array',
+      items: {
+        type: 'object',
+        properties: {
+          date: { type: 'string' },
+          author: { type: 'string' },
+          log: { type: 'string' },
+        },
+        required: ['date', 'author', 'log'],
+      },
+      nullable: true, // JSON schema treats undefined as nullable?
+    },
+    authorComments: {
+      type: 'string',
+      nullable: true, // JSON schema treats undefined as nullable?
+    },
+  },
+  /*
+    // the spec theoretically defines allowed license versions. Uncomment to enable this check
```

Comment:
We don't allow because it doesn't work, or because it's not really defined?

Reply:
It's not well defined. The spec (https://h5p.org/documentation/developers/json-file-definitions#guides-header-10) mixes up `license` and `licenseVersion`. We could enable it, but I don't think it's worth the hassle